### PR TITLE
Fix for pairtree prefix issue

### LINKED ISSUE #538: Adding works that don't have pairtree prefixes/version files in the directory causes 500 error
Discovered when trying to add uga1.32108002998303 to production, which is the first time I was adding a uga1 work to the database. 

@rlskoeser said that the directory for uga1.32108002998303 doesn't have the pairtree prefix and pairtree version files that are part of the spec and which the library we're using expects to be in place (HathiTrust doesn't use them). 

Potential solution is to create logic to create them (in the past we likely created them manually).

### PR DESCRIPTION
**Pull request by laurejt**
Fixes #538 for HathiTrust imports via the admin interface but not bulk imports.

### CODE REVIEW

File: ppa/archive/hathi.py [hunk 1]
```diff
@@ -269,9 +269,32 @@ def content_dir(self):
     def pairtree_client(self):
         """Initialize a pairtree client for the pairtree datastore this
         object belongs to, based on its Hathi prefix id."""
+        store_dir = os.path.join(settings.HATHI_DATA, self.pairtree_prefix)
+        ptree_file_prefix = f"{self.pairtree_prefix}."
```
Comment (rlskoeser) [inline ppa/archive/hathi.py:273]: What do you think about moving the logic and correcting the behavior of the `self.pairtree_prefix` property so it includes the '.' as it should? You'll have to update a few unit tests, but it looks like it is only used here. I guess you'd then need to update the logic to remove it when generating the path for the `store_dir`.

Defer to you on which is preferable, I'm fine with either.

**Review by rlskoeser — APPROVED** [review]
This looks great.

I think the scripts for bulk hathi add/import may be fine. The `hathi_import` script is meant to work from an existing pairtree structure on disk, and uses it to populate the digitized works in the database. There's a separate `hathi_add` script that is equivalent to the admin hathi add/import functionality (inconsistent terminology, sorry!), and that one uses the code that you've fixed.

I'll plan to do brief testing of the scripts when Mary tests the admin version so we can determine what works and document it. Depending what we find, if we think it might cause developer headaches in the future then we can figure out somewhere useful to document (or add a helpful error message to the script).

File: ppa/archive/hathi.py [hunk 2]
```diff
@@ -242,58 +242,46 @@ class HathiObject:
...
-    @cached_property
-    def content_dir(self):
-        """content directory for this work within the appropriate
-        pairtree"""
-        # contents are stored in a directory named based on a
-        # pairtree encoded version of the id
-        return pairtree_path.id_encode(self.pairtree_id)
+        # Identifiers for owning institution and volume which form the overall
+        # HathiTrust record id: [lib_id].[vol_id]
+        self.lib_id, self.vol_id = hathi_id.split(".", 1)
```
Comment (rlskoeser) [inline ppa/archive/hathi.py:256]: nice!

File: ppa/archive/hathi.py [hunk 3]
```diff
@@ -242,58 +242,46 @@ class HathiObject:
     """An object for working with a HathiTrust item with data in a
     locally configured pairtree datastore."""
 
-    hathi_id = None
+    # Pairtree version statement usd by pairtree package
+    pairtree_version_stmt = (
+        "This directory conforms to Pairtree Version 0.1. Updated spec: " +
+        "http://www.cdlib.org/inside/diglib/pairtree/pairtreespec.html"
+        )
```
Comment (rlskoeser) [inline ppa/archive/hathi.py:249]: 👍

**Comment by laurejt** [comment]
@rlskoeser Thanks for the initial review. I've refactored the code so that more of the logic is within the class itself. I also did some additional code clean-up to convert some cached properties into class attributes, but only for `HathiObject`.

File: ppa/archive/tests/test_hathi.py [hunk 4]
```diff
@@ -222,43 +222,41 @@ class TestHathiObject:
...
+    def test_vol_id(self):
         hobj = hathi.HathiObject(hathi_id="uva.1234")
-        assert hobj.pairtree_prefix == "uva"
+        assert hobj.vol_id == "1234"
 
-    def test_pairtree_id(self):
+    def test_pairtree_prefix(self):
         hobj = hathi.HathiObject(hathi_id="uva.1234")
-        assert hobj.pairtree_id == "1234"
+        assert hobj.pairtree_prefix == "uva."
```
Comment (rlskoeser) [inline ppa/archive/tests/test_hathi.py:235]: these could all be tested in a `test_init` method now that they aren't implemented as separate properties (but it's fine to leave as is!)

**Review by rlskoeser — APPROVED** [review]
Nice refactor! So much cleaner and simpler

File: ppa/archive/tests/test_hathi.py (hunk 4 above)
Comment (laurejt) [reply ppa/archive/tests/test_hathi.py:235]: I was debating this, I might as well make the change while I'm cleaning up this particular section.

**Review by rlskoeser — APPROVED** [review]

**Comment by laurejt** [comment]
@rlskoeser Thanks for the follow-up review! This should be the last review request; I just refactored the tests to make a single initialization test for `HathiObject`.

**Pull request merged**